extensions/postgresql-metadata-storage/src/main/java/io/druid/metadata/storage/postgresql/PostgreSQLConnector.java

```diff
@@ -112,4 +114,16 @@ public Void withHandle(Handle handle) throws Exception
 
   @Override
   public DBI getDBI() { return dbi; }
+
+  @Override
+  protected boolean isTransientException(Throwable e)
+  {
+    if(e instanceof SQLException) {
+      final String sqlState = ((SQLException) e).getSQLState();
+      // retry on connection errors and insufficient resources
+      // (see http://www.postgresql.org/docs/current/static/errcodes-appendix.html)
+      return sqlState.startsWith("08") || sqlState.startsWith("53");
```

Comment:
I'm thinking more generally in terms of things that are not the client's fault and that might get resolved within a few retries, either manually or without intervention.